pkg/osutil/interrupt_unix.go

```diff
@@ -60,9 +60,7 @@ func HandleInterrupts(lg *zap.Logger) {
 
 		interruptExitMu.Lock()
 
-		if lg != nil {
```

Comment:
If the `lg` shouldn't be nil, then it would be better to add an assertion. 

FYI. https://github.com/etcd-io/etcd/blob/4d57eb8d3b2190bfdba3c65f8eb93c0349fc6dcc/pkg/schedule/schedule.go#L92

Reply:
It‘s  should not be nil because it would be checked in https://github.com/etcd-io/etcd/blob/4d57eb8d3b2190bfdba3c65f8eb93c0349fc6dcc/server/etcdmain/etcd.go#L54. And then call `HandleInterrupts` in https://github.com/etcd-io/etcd/blob/4d57eb8d3b2190bfdba3c65f8eb93c0349fc6dcc/server/etcdmain/etcd.go#L182.

Reply:
OK, it would be better to an assertion